docs/examples/partitioning/jobs/function-num-columns-changed.sql

```diff
@@ -0,0 +1,22 @@
+
+-- Create function to allow determination if a schema has changed
+
+CREATE OR ALTER FUNCTION NUM_COLUMNS_CHANGED(@t1 VARCHAR(255), @t2 VARCHAR(255))
```

Comment:
Maybe a future ticket about doing a more detailed schema comparsion? I suspect that many times a schema change would only be the change of a datatype of a column... or potentially adding/removing columns at the same time and still ending up with the same number of columns... would it be straightforward to come up with a true/false type query based on something like this (would likely be database specific, but I think that's fine): https://www.mssqltips.com/sqlservertip/4824/easy-way-to-compare-sql-server-table-schemas/

Reply:
Sure.  This is just example code, it can be adapted and adjusted as implementers see fit.  To be clear, this function _does_ only look for datatype changes and name changes, or additions/deletions of columns.  It doesn't worry about other aspects of schema changes.  And I originally implemented this as a boolean function, but it ended up not being something I could use without comparing to 0, so I ended up making it a numeric return value to make this more explicit.